base/random/RNGs.jl

```diff
@@ -257,12 +257,14 @@ rand(r::MersenneTwister, ::Type{Int128}) = reinterpret(Int128, rand(r, UInt128))
 #### arrays of floats
 
 function rand_AbstractArray_Float64!(r::MersenneTwister, A::AbstractArray{Float64},
-                                     n=length(A), I::FloatInterval_64=CloseOpen())
+                                     region::AbstractUnitRange, I::FloatInterval_64)
+    # region should be a subset of linearindices(A)
     # what follows is equivalent to this simple loop but more efficient:
-    # for i=1:n
+    # for i=region
     #     @inbounds A[i] = rand(r, I)
     # end
-    m = 0
+    m = Base.checked_sub(first(region), 1)
+    n = last(region)
```

Comment:
Shouldn't it check somewhere that `m` and `n` are in `indices(A)`, in order for `@inbounds` to be valid?

Reply:
Or is it because `rand_AbstractArray_Float64!` is only supposed to be called internally by code where `region` has already been checked?  (But maybe there should be an `assert` in that case?)

Reply:
Yes it's an internal function so `@inbounds` should be always correct, but an `@assert` can't hurt here either.